src/libclipper/include/clipper/datatypes.hpp

```diff
@@ -17,6 +17,7 @@ using QueryId = long;
 using FeedbackAck = bool;
 
 enum class InputType {
+  Invalid = -1,
```

Comment:
I'm assuming you needed to add this for a reason? Why do we need an invalid type?